airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/JsonSchemaType.java

```diff
@@ -16,8 +16,8 @@
 public enum JsonSchemaType {
 
   STRING("string", true, null, Schema.Type.STRING),
-  NUMBER_INT("number", true, "integer", Schema.Type.INT),
-  NUMBER_LONG("number", true, "big_integer", Schema.Type.LONG),
+  NUMBER_INT("number", true, "integer", Schema.Type.LONG),
+  NUMBER_LONG("number", true, "long", Schema.Type.STRING),
```

Comment:
@sashaNeshcheret are there already sources which produce `big_integer` as meaning an int64?

there's a project in-planning to change how we handle these types in general, so I'm not _too_ worried about long-term implications. Just want to make sure we don't break anything in the short/mid term.

Reply:
I'd like to reserve big_integer for an unbounded int (i.e. something similar to java's BigInteger class). Eventually I think we'll have something like this:
* `{type: ???, airbyte_type: int}` -> avro int
* `{type: ???, airbyte_type: long}` -> avro long
* `{type: ???, airbyte_type: big_integer}` -> avro string (? not sure if avro has an actual unbounded int type)

but we're still figuring out the details.

Reply:
Got your point regarding unbounded int, but such approach has disadvantages that we will lose type for all sources that really has range as big_integer and instead of using Schema.Type.LONG we will use Schema.Type.String. I believe we can add additional enum value for such case - NUMBER_BIG_INTEGER("number", true, "big_integer", Schema.Type.STRING), and map sources data types with range more than Long value to NUMBER_BIG_INTEGER and other to NUMBER_LONG, wdyt?